h/streamer/streamer.py

```diff
@@ -56,27 +63,22 @@ def process_work_queue(settings, queue, session_factory=None):
     code that ensures the database session is appropriately committed and
     closed between messages.
     """
-    if session_factory is None:
-        session_factory = _get_session
-    session = session_factory(settings)
-    topic_handlers = {
-        ANNOTATION_TOPIC: messages.handle_annotation_event,
-        USER_TOPIC: messages.handle_user_event,
-    }
+
+    session = db.Session(bind=db.make_engine(registry.settings))
```

Comment:
Get rid of this so there's no branching. This injection was only for the tests. We just mock `db` now.